samples/ADC/Program.cs

```diff
@@ -14,37 +13,37 @@ public class Program
     {
         public static void Main()
         {
-            string devs = AdcController.GetDeviceSelector();
+            // nanoFramework provides a single "virtual" ADC controller, even if the target has multiple ones.
+            AdcController adc = AdcController.GetDefault();
 
-            Debug.WriteLine("devs=" + devs);
+            // Minimum value is usually 0, but maximum depends on the board and the actual physical analog-to-digital converter.
+            // For example, STM32F769 has 12bit ADC, so the maximum value is 4095.
+            int maximum = adc.MaxValue;
+            int minimum = adc.MinValue;
+            Debug.WriteLine($"ADC minimum value is {minimum}, maximum - {maximum}");
 
-            AdcController adc1 = AdcController.GetDefault();
+            // Analog channels in nanoFramework are defined sequentially: 0, 1, 2, etc. However, these number are mapped inside firmware,
```

Comment:
It would be helpful to point out that there are board support packages for almost all targets, on which the developer can find an helper class that maps and documents the ADC channels. 
[NuGet for ST_STM32F769I_DISCOVERY](https://www.nuget.org/packages/nanoFramework.Targets.ST_STM32F769I_DISCOVERY/).

Pointing to the various reference targets pages it's a good suggestion too.

Reply:
"Almost all" is they key here. Almost is not all. In fact, if I search for "nanoFramework.Targets.", it gives me 5 results. 5 is less than a half of total _official_ targets. Also, those Nugets...  "Project Site" on the NuGet.org links to the landing page.  They do not have repositories, or at least I cannot find them. So I don't even know what's inside them.

So at this point, this whole "board support package" thing needs revision before it can be recommended in samples...

Reply:
There are BSPs for the targets that someone took the time to code for. No one else care to code for the rest of them. That doesn't make the ones that exist less relevant or useless. 
Pretty much inline with the pages with the details for each target. They are empty for ...some... of the targets. Others are pretty complete. Others not so much. That doesn't make the ones that exist less relevant or useless. 

Just improvement my comment above, hopefully with satisfactorily enough accuracy...

Those nugets point to the GitHub organization on purpose. Has a global pointer to the project.

BTW, the code that's packaged on those lives inside the folder named `managed_helpers` inside the respective target folder in the nf-interpreter repository. 

In case you're not aware of it, to explore what's inside a NuGet package you don't need to download or install the package. Just click in the "Open in NuGet Package Explorer" URL, right there on the nuget.org page for the package.
![image](https://user-images.githubusercontent.com/1881520/133602882-d9c83685-8f86-4603-8d44-b1b07ddef627.png)

Reply:
Ok so let's sum this up. You insist on adding links to NuGets, which are:
- Only for certain boards;
- Updated more than two years ago;
- Functionally not very complete (only contains ADC mappings);
- Without proper Source Repository links (content's real source has to be reverse engineered);
- Completely ignored by nF owners themselves (GitHub returns zero occurencies in packages.config file content):
https://github.com/search?q=org%3Ananoframework+STM32F429I_DISCOVERY+extension%3Aconfig&type=Code&ref=advsearch&l=&l=

Having said all that, and given the general vibe of Samples repository not being up to the snuff, I have 0 reasons to believe those NuGets actually are correct. Sorry, I will not be buying new boards just to check that. Netduino is not even sold anymore...

Yet you still prefer to keep a broken sample online, although there's not a single usage of nanoFramework.Targets.* NuGet in the entire nF organization.

I mean, that's your project, you're the boss and I will respect that, but I really don't understand such pickiness. Apparently, I'm not the contributor you're looking for.

Reply:
I was simply pointing out the value of the BSPs that exist. 

The fact that those were updated 2 years ago, doesn't meant they are outdated. They date back from the time the targets where created and since then the ADC channel assignment hasn't changed. Thus no need to update the helper classes.

This is a sample for ADC, seems logical to address ADC related stuff. If those BSPs could have more helper classes? Sure they could.

Even if the project repository link would point to the repository, it still would require an extra pointer to find the source files included in the package.

The BSP's are supposed to be used by the end users, so there is not much surprise that you don't see any usage in the project repos. The only potential candidates for that would be samples, but only for samples targeting a very specific target board, which kind of goes against the "generic demo" approach that's being followed on the samples.

Wasn't expecting, at all, nor I had suggested that you would be buying those boards to check the samples or NuGets.

For clarity sake: I haven't stated anywhere that I prefer to have broken samples. Making this depend on the analysis of the usage of nanoFramework.Targets.* NuGets seems a bit far fetched.

Really sorry that you see these review comments as a pickiness from my side... Just like you've contributed with a PR to improve the sample in the way you saw most fit, I have commented on those changes with the exact same goal of further improving it, in order to have a better sample. That's the usual and expected workflow.

All contributions matter and we welcome everyone willing to positively contribute to the project, according to the [Code of Conduct](https://dotnetfoundation.org/about/code-of-conduct).